front/lib/api/agent_data_sources.ts

```diff
@@ -73,3 +73,53 @@ export async function getDataSourcesUsageByAgents({
     {}
   );
 }
+
+export async function getDataSourceUsage({
+  auth,
+  dataSourceId,
+}: {
+  auth: Authenticator;
+  dataSourceId: number;
```

Comment:
First you want to use a ModelId but we have an invariant as per the engineering notion page that we don't expose ModelId in lib/api signatures. Please use DataSourceType here